doc/rst/developer/bestPractices/Deprecation.rst

```diff
@@ -73,3 +73,201 @@ Optionally include deprecation timeline in comments of deprecation test:
+be tricky for a user to opt in to the new behavior.
+
+.. code-block:: chapel
+
+   deprecated "foo returning 'int' is deprecated"
+   proc foo(): int { ... }
+
+   // As written, this replacement would conflict with the original, how can a
+   // user call it?
+   proc foo(): bool { ... }
+
+In this situation, we recommend adding a ``config param`` and a ``where`` clause
+that responds to it to the deprecated function and its replacement:
+
+.. code-block:: chapel
+
+   // The default state should result in the deprecated behavior, so users can
+   // adjust their code at their leisure
+   config param fooReturnsBool = false;
+
+   // The deprecation message should alert the user to the new flag
+   deprecated "foo returning 'int' is deprecated, please compile with '-sfooReturnsBool=true' to get the new return type"
+   proc foo(): int where !fooReturnsBool { ... }
+
+   // The new function should also use a 'where' clause to opt in to the new
+   // behavior
+   proc foo(): bool where fooReturnsBool { ... }
+
+When the deprecated function is removed, the flag should also be deprecated (and
+removed from the new function to avoid generating noise for the user):
+
+.. code-block:: chapel
+
+   deprecated "'fooReturnsBool' is deprecated and no longer has an effect"
+   config param fooReturnsBool = false;
+
+   // The old version has been removed, and the flag is no longer needed, so
+   // the new function can be in its final state.
+   proc foo(): bool { ... }
+
+There is a drawback with this approach - config params are not adjustable during
+the lifetime of a program, so updates must be made all at once before they can
+be tested by the user.  So, if you can think of a better solution, please
+suggest it and we will update this example!
+
+Changing A Function's Argument Name
++++++++++++++++++++++++++++++++++++
+
+When only the name of a function argument needs to change and not its type, a
+new overload will encounter conflicts when a user relies solely on positional
+ordering:
+
+.. code-block:: chapel
+
+   deprecated "argument name 'a' is deprecated, use 'b' instead"
+   proc foo(a: int) { ... }
+
+   proc foo(b: int) { ... }
+
+   ...
+
+   // This will obviously use the deprecated version
+   foo(a=3);
+   // And this will obviously use the new version
+   foo(b=3);
+   // But which overload will this call?  The compiler can't choose between them
+   foo(3);
+
+In this case, we still want to generate warnings when the old argument name is
+used, but we want positional ordering to work without indicating anything has
+changed.  To accomplish this, mark the deprecated version with ``pragma "last
+resort"`` - this will avoid conflicts in the positional ordering case while
+still keeping the old argument name available to generate the deprecation
+warning:
+
+.. code-block:: chapel
+
+   pragma "last resort"
+   deprecated "argument name 'a' is deprecated, use 'b' instead"
+   proc foo(a: int) { ... }
```

Comment:
I think we can expand this to 
```
proc foo(a: int) {
  foo(a)
}
```
to emphasize we don't need to have the contents of `foo` in both the deprecated function (`proc foo(a: int)`) and the new function (`proc foo(b: int)`)

Reply:
Done, does that look good?

Reply:
Looks good, thanks!